src/Commands/Admin/GetTenantInfo.cs

```diff
@@ -0,0 +1,53 @@
+﻿using PnP.Core.Transformation;
+using PnP.PowerShell.Commands.Base;
+using PnP.PowerShell.Commands.Model;
+using PnP.PowerShell.Commands.Utilities.REST;
+using System.Management.Automation;
+
+namespace PnP.PowerShell.Commands.Admin
+{
+    [Cmdlet(VerbsCommon.Get, "PnPTenantInfo")]
+    public class GetTenantInfo : PnPAdminCmdlet
+    {
+        [Parameter(Mandatory = false)]
+        public string DomainName;
+
+        [Parameter(Mandatory = false)]
+        public string TenantId;
+
+        protected override void ExecuteCmdlet()
+        {
+            EnsureValidInput();
+
+            WriteVerbose("Acquiring access token for Microsoft Graph to look up Tenant");
+            var graphAccessToken = TokenHandler.GetAccessToken(this, $"https://{Connection.GraphEndPoint}/.default", Connection);
+            var requestUrl = BuildRequestUrl();
+
+            var results = RestHelper.GetAsync<Model.TenantInfo>(Connection.HttpClient, requestUrl, graphAccessToken).GetAwaiter().GetResult();
+            WriteObject(results, true);
+        }
+
+        private void EnsureValidInput()
+        {
+            if (string.IsNullOrEmpty(TenantId) && string.IsNullOrEmpty(DomainName))
```

Comment:
You should use ParameterSets for this instead of this check, please check other cmdlets for example.

Reply:
Hey @gautamdsheth 

Suggested changes are done accordingly.

Thanks,
Nishkalank Bezawada